feature_engine/encoding/decision_tree.py

```diff
@@ -163,6 +169,15 @@ def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
             ordered ordinal encoding.
         """
 
+        # confirm model type and target variables are compatible.
+        if self.regression is True and type_of_target(y) == "binary":
+            raise ValueError(f"'regression' is {self.regression} and target "
+                             f"variables are a binary. These two parameter settings "
+                             f"are not compatible.")
```

Comment:
Can we re-phrase the error message to something like this:

"Trying to fit a regression to a binary target is not allowed by this transformer.
Check the target values or set regression to False."